kn/fotos/entities.py

```diff
@@ -269,42 +282,26 @@ def update_metadata(self):
             if 'DateTimeOriginal' in exif:
                 self.created = datetime.datetime.strptime(exif['DateTimeOriginal'], '%Y:%m:%d %H:%M:%S')
 
+        if self.size is None:
+            self.size = img.size
+
+        return True
 
     def _cache(self, cache):
         if cache == 'full':
-            return {}
+            return
         source = self.original_path
         target = self.get_cache_path(cache)
         target_dir = os.path.dirname(target)
         if not os.path.exists(target_dir):
             os.makedirs(target_dir)
-        if cache == 'thumb':
-            subprocess.check_call(['convert',
-                             source,
-                             '-resize', '200x',
-                             '-rotate', str(self.rotation),
-                             target])
-        elif cache == 'thumb2x':
-            subprocess.check_call(['convert',
-                             source,
-                             '-resize', '400x',
-                             '-rotate', str(self.rotation),
-                             target])
-        elif cache == 'large':
-            subprocess.check_call(['convert',
-                             source,
-                             '-resize', '850x',
-                             '-rotate', str(self.rotation),
-                             target])
-        elif cache == 'large2x':
-            subprocess.check_call(['convert',
-                             source,
-                             '-quality', '90',
-                             '-resize', '1700x',
-                             '-rotate', str(self.rotation),
-                             target])
-        # No worries: Image.open is lazy and will only read headers
-        return {'size': Image.open(target).size}
+
```

Comment:
Oke, dat is dan goed.

Van de [ImageMagick documentatie](http://www.imagemagick.org/script/command-line-options.php#quality):
> For the JPEG and MPEG image formats, quality is 1 (lowest image quality and highest compression) to 100 (best quality but least effective compression). The default is to use the estimated quality of your input image if it can be determined, otherwise 92. When the quality is greater than 90, then the chroma channels are not downsampled. Use the -sampling-factor option to specify the factors for chroma downsampling.

... dus ik denk dat andere foto's kwaliteit 92 gebruiken.

Wat zullen we doen? Alle foto's op kwaliteit 90 zetten? Thumbnails lagere kwaliteit geven? Of alleen de `large2x` op kwaliteit 90 zetten?
Mijn voorkeur gaat uit naar optie 2 (`thumb*` 85 en `large*` 90).